core/domain/user_services.py

```diff
@@ -111,7 +119,27 @@ def require_valid_username(cls, username):
             # Admin usernames are reserved for admins. Note that 'admin'
             # itself is already in use for the demo exploration.
             raise utils.ValidationError('This username is not available.')
-
+    @classmethod
+    def fetch_gravatar(cls, email):
```

Comment:
Oh, wait, sorry, I take that back. I got confused by the docstring below -- I confused this with the autogenerated icons.

The gravatar is not really an autogenerated avatar (which we use as a representation of the user when their avatar is unknown). It's a user's actual avatar. I suggest amending the docstring to "Gets the gravatar for a given email." If no gravatar can be found, return None.